Fix shellcheck failures in test/cmd/{d,l}.*.sh

xychu opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. Follow the instructions for writing a release note: https://git.k8s.io/community/contributors/guide/release-notes.md
6. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Part of #72956
**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->
Fixes #

**Special notes for your reviewer**:
This is a follow-up pr of #74639 and it is part of fixes under test/cmd/*.sh
**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".
-->
```release-note
NONE
```

xychu committed:
Fix shellcheck failures in test/cmd/d.*.sh

xychu committed:
Fix shellcheck failures in test/cmd/l.*.sh

xychu commented:
/assign @BenTheElder 
Hope this will be the last shellcheck failures fix under "/test/cmd/"

BenTheElder commented on test/cmd/legacy-script.sh:
```diff
@@ -337,43 +337,43 @@ runTests() {
+  export template_generation_field=".spec.templateGeneration"
+  export container_len="(len .spec.template.spec.containers)"
+  export image_field0="(index .spec.template.spec.containers 0).image"
+  export image_field1="(index .spec.template.spec.containers 1).image"
```
why do we have so many exports here?

xychu replied:
Check failed with "SC2034: image_field1 appears unused. Verify use (or export if used externally)."
All of them are used externally and I think export them should be the case we need.

BenTheElder replied:
where are they used?

xychu replied:
They are used in some other files under `test/cmd/`.
``` base
# grep "image_field1" -rn test/cmd/*
test/cmd/apps.sh:78:  kube::test::wait_object_assert daemonset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_DAEMONSET_R2_2}:"
test/cmd/apps.sh:84:  kube::test::get_object_assert daemonset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_DAEMONSET_R2_2}:"
test/cmd/apps.sh:98:  kube::test::wait_object_assert daemonset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_DAEMONSET_R2_2}:"
test/cmd/apps.sh:339:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
test/cmd/apps.sh:343:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
test/cmd/apps.sh:349:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
test/cmd/apps.sh:353:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
test/cmd/apps.sh:357:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
test/cmd/apps.sh:361:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_DEPLOYMENT_R1}:"
test/cmd/apps.sh:365:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_DEPLOYMENT_R1}:"
test/cmd/apps.sh:430:  kube::test::wait_object_assert statefulset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PAUSE_V2}:"
test/cmd/apps.sh:436:  kube::test::get_object_assert statefulset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PAUSE_V2}:"
test/cmd/apps.sh:450:  kube::test::wait_object_assert statefulset "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PAUSE_V2}:"
test/cmd/core.sh:1267:  kube::test::get_object_assert deployment "{{range.items}}{{${image_field1:?}}}:{{end}}" "${IMAGE_PERL}:"
```
If it make more sense that I will add one comments above these declares to explain why we need to export them.

BenTheElder replied:
thanks!

sorry I'm so behind reviewing this

BenTheElder commented on test/cmd/legacy-script.sh:
```diff
@@ -67,14 +67,14 @@ CTLRMGR_PORT=${CTLRMGR_PORT:-10252}
 PROXY_HOST=127.0.0.1 # kubectl only serves on localhost.
 
 IMAGE_NGINX="k8s.gcr.io/nginx:1.7.9"
```
is this one unused?

xychu replied:
No, actually it's used in current file.
And because of this it's okay to not export it.

BenTheElder commented on test/cmd/legacy-script.sh:
```diff
@@ -98,10 +98,10 @@ secrets="secrets"
 services="services"
 statefulsets="statefulsets"
-static="static"
+export static="static"
```
this is a pretty goofy thing to export, what is reading this?

xychu replied:
Yeah, I got it. I've removed these kind of external use in last commit.

BenTheElder commented on test/cmd/legacy-script.sh:
```diff
@@ -98,10 +98,10 @@ secrets="secrets"
 storageclass="storageclass"
 subjectaccessreviews="subjectaccessreviews"
-selfsubjectaccessreviews="selfsubjectaccessreviews"
+export selfsubjectaccessreviews="selfsubjectaccessreviews"
```
same here

xychu replied:
Cleaned up as well.

xychu committed:
Fix of review comments. Will squash later.

Including:
- Remove external usage of resource name vars
- Add commments about why we export field helpers

BenTheElder commented on test/cmd/legacy-script.sh:
```diff
@@ -395,10 +394,10 @@ runTests() {
-      if [[ "${pkg}" != "kubeadm" ]]; then 
-        record_command run_${pkg}_tests
+      if [[ "${pkg}" != "kubeadm" ]]; then
+        record_command run_"${pkg}"_tests
```
nit: quote the whole string
```suggestion
        record_command "run_${pkg}_tests"
```

xychu replied:
Updated :-)

BenTheElder commented on test/cmd/legacy-script.sh:
```diff
@@ -324,7 +320,7 @@ runTests() {
   # token defined in hack/testdata/auth-tokens.csv
   kube_flags_with_token=(
-    -s "https://127.0.0.1:${SECURE_API_PORT}" --token=admin-token --insecure-skip-tls-verify=true
+    -s "https://127.0.0.1:${SECURE_API_PORT}" "--token=admin-token" "--insecure-skip-tls-verify=true"
```
perhaps -s should be quoted too? also if it's not to be expanded we should prefer single quotes:
```suggestion
    '-s' "https://127.0.0.1:${SECURE_API_PORT}" '--token=admin-token' '--insecure-skip-tls-verify=true'
```

xychu replied:
Same here

xychu committed:
Fix review comments, will squash later.

BenTheElder reviewed (approved):
/lgtm
/approve

BenTheElder commented:
/ok-to-test
/priority important-longterm
